### PR TITLE
Add .well-known option to control default e2ee behaviour

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/13705

image
image

### CODE REVIEW

**Comment by t3chguy** [comment]
> Show tile similar to the Encryption enabled tile for non-e2ee rooms

This has been descoped due to complexity with injecting tiles unrelated to timeline events and will be handled by next week's "Notifications" rework project.

File: src/widgets/Jitsi.ts [hunk 1]
```diff
@@ -36,33 +34,21 @@ export class Jitsi {
...
-            } catch (e) {
-                // These are non-fatal errors
-                console.error(e);
-            }
+        const cli = MatrixClientPeg.get();
+        const discoveryResponse = cli && MatrixClientPeg.get().getClientWellKnown();
+
+        if (cli) {
+            cli.on("WellKnown.client", () => this.update);
+        }
```
Comment (dbkr) [inline src/widgets/Jitsi.ts:46]: The update method is adding itself as a listener? How does it start listening in the first place, and does it leak listeners after that?

Comment (t3chguy) [reply src/widgets/Jitsi.ts:46]: True about the leakiness, the first update call is from Lifecycle, will split into start & update

**Comment by Bun-Bun** [comment]
Can this be expanded to include not only default behavior but force off behavior?

**Comment by t3chguy** [comment]
No. That is a distinct issue, as you know, which should be enforced by the server for consistent behaviour between clients.

**Pull request merged**